packages/@aws-cdk/aws-s3/lib/notifications-resource/lambda-source/src/index.py

```diff
@@ -0,0 +1,72 @@
+import boto3  # type: ignore
+import copy
+import json
+import urllib.request
+from typing import List
+
+s3 = boto3.client("s3")
+CONFIG_TYPES = ["TopicConfigurations", "QueueConfigurations", "LambdaFunctionConfigurations"]
+
+
+def handler(event: dict, context):
+    response_status = "SUCCESS"
+    error_message = ""
+    try:
+        props = event["ResourceProperties"]
+        bucket = props["BucketName"]
+        config = prepare_config(
+            current_config=s3.get_bucket_notification_configuration(Bucket=bucket),
+            in_config=props["NotificationConfiguration"],
+            old_config=event.get("OldResourceProperties", {}).get("NotificationConfiguration", {}),
+            merge=event["RequestType"] != "Delete",
+        )
+        s3.put_bucket_notification_configuration(Bucket=bucket, NotificationConfiguration=config)
+    except Exception as e:
+        print("Failed to put bucket notification configuration:", e)
+        response_status = "FAILED"
+        error_message = f"Error: {str(e)}. "
+    finally:
+        submit_response(event, context, response_status, error_message)
+
+
+def prepare_config(current_config: dict, in_config: dict, old_config: dict, merge: bool) -> dict:
+    config = copy.deepcopy(current_config)
+    config.pop("ResponseMetadata", None)
+    for config_type in CONFIG_TYPES:
+        config.setdefault(config_type, [])
+        in_config.setdefault(config_type, [])
+        filter_config(config, config_type, in_config, old_config)
+        if merge:
+            config[config_type].extend(in_config[config_type])
+    return config
+
+
+def filter_config(config: dict, config_type: str, in_config: dict, old_config: dict):
+    remove_ids = ids(in_config[config_type] + old_config.get(config_type, []))
+    config[config_type] = [item for item in config[config_type] if item["Id"] not in remove_ids]
```

Comment:
This is what i'm striving for:

```suggestion
    return [item for item in config[config_type] if item["Id"] not in remove_ids]
```

And then just make the assignment/mutation on the caller side, that actually created the `config` dictionary.

Reply:
@iliapolo  updated